pkg/azure/ipam/node.go

```diff
@@ -197,28 +211,39 @@ func (n *Node) ResyncInterfacesAndIPs(ctx context.Context, scopedLog *logrus.Ent
 	return available, stats, nil
 }
 
-// GetMaximumAllocatableIPv4 returns the maximum amount of IPv4 addresses
-// that can be allocated to the instance
-func (n *Node) GetMaximumAllocatableIPv4() int {
+// GetMaximumAllocatableIP returns the maximum amount of IP addresses
+// that can be allocated to the instance based on the provided IP family.
+func (n *Node) GetMaximumAllocatableIP(family ipam.Family) int {
+	if family == ipam.IPv6 {
+		// not implemented
+		return 0
+	}
```

Comment:
It doesn't seem like [Azure distinguishes](https://github.com/MicrosoftDocs/azure-docs/blob/main/includes/azure-virtual-network-limits.md#networking-limits---azure-resource-manager) between ip4/ip6 for this limit, so we can probably just leave this as is and make a note for the follow metrics work.

Reply:
If the `family` parameter is removed, the Azure `Node` type will not not implement the IPAM `NodeOperations` interface.